bmc/mock/main.go

```diff
@@ -0,0 +1,41 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	"github.com/ironcore-dev/metal-operator/bmc/mock/server"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func main() {
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
+	defer stop()
+
+	opts := zap.Options{
+		Development: true,
+	}
+
+	log := ctrl.Log.WithName("RedfishMockServer")
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	srv := server.NewMockServer(log, ":8000")
+
+	if err := srv.Start(ctx); err != nil {
+		log.Error(err, "Failed to start mock server")
+		return
+	}
+
+	<-ctx.Done()
+
+	time.Sleep(1 * time.Second)
```

Comment:
Context handling seems a bit iffy to me. Something like the following across `main.go` and `server.go` should suffice:
```go
ctx, cancel := signal.NotifyContext()
defer cancel()

go server.ListenAndServe()

<-ctx.Done()
timeout, cancel := context.WithTimeout(context.Background(), time.Second)
defer cancel()
server.Shutdown(timeout)
```

Reply:
Wall my idea was to have a cancelable `Start()` method which I can use in the suite test setup and stop the server via `defer` cancelling the context. With the suggested solution I need to replicate what has been done in `main.go` in every suite test setup or?